core/src/main/java/org/infinispan/stats/impl/StatsImpl.java

```diff
@@ -47,7 +49,30 @@
    final CacheMgmtInterceptor mgmtInterceptor;
    final Stats source;
 
-   public StatsImpl(AsyncInterceptorChain chain) {
+   public static final class StatsFactory {
+
+      public static final Stats create(Configuration configuration, AsyncInterceptorChain chain) {
+         if (!configuration.jmxStatistics().available()) {
+            throw new CacheException("JMX Statistics are not available");
```

Comment:
Why throw an exception here, while in the other ```create``` we silently cope with disabled jmx stats by providing empty statistics. Can we have a uniform approach?

Reply:
@wburns suggested this behaviour

Reply:
@wburns suggested this behaviour

Reply:
Ok, now we know who to blame :) How does the defendant plead?

Reply:
I didn't say to have different behavior. All I said was we could throw an exception. I agree that they should be consistent. Whether we want to have an exception or empty statistics is up for discussion though. I personally am not the biggest fan of having stats that just say 0, that could be a valid stat :)

Reply:
some stats say -1, instead of 0, don't they

Reply:
@wburns do we return -1 stats then ?

Reply:
Ah, I missed that `emptyStats` puts a -1. Looking more closely, that is what we did in some cases so I would just say to keep returning -1 then.